### PR TITLE
Syntax Corrections

### PR DESCRIPTION
**Pull request by JonTheWong**
Corrections for markdown, changed string for MD5, changed requirements
Added SHA example

### CODE REVIEW

File: applications/crossbar/doc/user_authentication.md [hunk 1]
```diff
@@ -12,10 +12,10 @@ Provides an auth-token via user credentials
 
 Key | Description | Type | Default | Required | Support Level
 --- | ----------- | ---- | ------- | -------- | -------------
-`account_name` | The account name of the user | `string(1..128)` |   | `false` |  
+`account_name` | The account name of the user | `string(ABC..Z)` |   | `true` |
```
Comment (jamesaimonetti) [inline applications/crossbar/doc/user_authentication.md]: `string(1..128)` just notes that the account name can be up to 128 characters. It can contain alphanumeric values too so `A-Z` wouldn't be accurate anyway.

Additionally the tables are auto-generated from the JSON schemas so editing them here in markdown is incorrect - find the corresponding JSON schema and edit the "description".

File: applications/crossbar/doc/user_authentication.md [hunk 2]
```diff
@@ -12,10 +12,10 @@ Provides an auth-token via user credentials
 
 Key | Description | Type | Default | Required | Support Level
 --- | ----------- | ---- | ------- | -------- | -------------
-`account_name` | The account name of the user | `string(1..128)` |   | `false` |  
+`account_name` | The account name of the user | `string(ABC..Z)` |   | `true` |  
 `account_realm` | The account realm of the user | `string(4..253)` |   | `false` |  
-`credentials` | A hash of the uses credentials | `string(1..64)` |   | `true` |  
-`method` | The hash method | `string('md5' | 'sha')` | `md5` | `false` |  
+`credentials` | A hash of the user credentials | `string(1..64)` |   | `true` |  
+`method` | The hash method | `string(md5..sha)` | `md5` | `true` |
```
Comment (jamesaimonetti) [inline applications/crossbar/doc/user_authentication.md]: `string(md5 | sha)` is how we denote that the value of `method` can be either `md5` OR `sha`, `..` indicating a range of values, `|` indicating an `OR`.

Comment (JonTheWong) [reply applications/crossbar/doc/user_authentication.md]: I added the .. to keep the style the same, only reason i removed the | operator is that it doesn't show properly when previewing. In this case the OR should have been used.

File: applications/crossbar/doc/user_authentication.md (hunk 1 above)
Comment (JonTheWong) [reply applications/crossbar/doc/user_authentication.md]: I understand, my reasoning was that it could be alphanumeric and it wasn't clear to us. I only read this part of the doc and should have done more research before submitting.

File: applications/crossbar/doc/user_authentication.md (hunk 2 above)
Comment (jamesaimonetti) [reply applications/crossbar/doc/user_authentication.md]: Yeah Github won't render everything exactly right. The docs site (see https://docs.2600hz.com/dev/applications/crossbar/doc/user_authentication/ for instance) uses mkdocs to render the markdown into HTML so that's the rendering we target to make work.

**Comment by jamesaimonetti** [comment]
@JonTheWong any timeline on updating the PR?

File: applications/crossbar/doc/user_authentication.md (hunk 2 above)
Comment (JonTheWong) [reply applications/crossbar/doc/user_authentication.md]: I understand.

File: applications/crossbar/doc/user_authentication.md [hunk 3]
```diff
@@ -12,9 +12,9 @@ Provides an auth-token via user credentials
 
 Key | Description | Type | Default | Required | Support Level
 --- | ----------- | ---- | ------- | -------- | -------------
-`account_name` | The account name of the user | `string(1..128)` |   | `false` |  
+`account_name` | The account name of the user | `string(1..128)` |   | `true` |
```
Comment (jamesaimonetti) [inline applications/crossbar/doc/user_authentication.md]: You are trying to change `support level` which is only a field for 2600Hz (and is populated based on the schema. Revert this change.
Also, `account_name` is not strictly required; `account_realm` or a `phone_number` assigned to the account serve the same purposes.

File: applications/crossbar/doc/user_authentication.md [hunk 4]
```diff
@@ -116,7 +123,7 @@ Send a reset-your-password email via user credentials
 
 Key | Description | Type | Default | Required | Support Level
 --- | ----------- | ---- | ------- | -------- | -------------
-`account_name` | The account name of the user | `string(1..64)` |   | `false` |
+`account_name` | The account name of the user | `string(ABC...Z)` |   | `false` |
```
Comment (jamesaimonetti) [inline applications/crossbar/doc/user_authentication.md]: Revert this

File: applications/crossbar/doc/user_authentication.md [hunk 5]
```diff
@@ -12,9 +12,9 @@ Provides an auth-token via user credentials
 
 Key | Description | Type | Default | Required | Support Level
 --- | ----------- | ---- | ------- | -------- | -------------
-`account_name` | The account name of the user | `string(1..128)` |   | `false` |  
+`account_name` | The account name of the user | `string(1..128)` |   | `true` |  
 `account_realm` | The account realm of the user | `string(4..253)` |   | `false` |  
-`credentials` | A hash of the uses credentials | `string(1..64)` |   | `true` |  
+`credentials` | A hash of the user credentials | `string(1..64)` |   | `true` |
```
Comment (jamesaimonetti) [inline applications/crossbar/doc/user_authentication.md]: This should be fixed in crossbar/priv/couchdb/schemas/user_auth.json, then run `make apis` from the root which should auto-update this table.
